lib/populator.rb

```diff
@@ -1,56 +1,68 @@
-module Populator
-  # singleton
-  # https://practicingruby.com/articles/ruby-and-the-singleton-pattern-dont-get-along
-  extend self
-
-  def uniform_ncr_data(n=10)
-    n.times do |i|
-      FactoryGirl.create(:ncr_work_order, :with_approvers)
+class Populator
+  include FactoryGirl::Syntax::Methods
+
+  def uniform_ncr_data(n: 10)
+    n.times do
+      create(:ncr_work_order, :with_approvers)
     end
   end
 
-  def random_bool(pct_true=0.5)
-    rand < pct_true
+  def random_ncr_data(num_proposals: 50)
+    num_proposals.times do
+      requested_at = rand(3.months.ago..1.day.ago)
+      proposal = create_proposal(requested_at: requested_at)
+      create_work_order(proposal: proposal)
+      add_comments(proposal: proposal, requested_at: requested_at)
+    end
   end
 
-  def random_ncr_data(num_proposals=50)
-    # all of the hard-coded numbers within here are fairly arbitrary
+  def ncr_data_for_user(email:, num_proposals: 25)
+    user = User.find_by(email_address: email)
 
-    num_proposals.times do |i|
-      requested_at = rand(3.months.ago..1.day.ago)
+    num_proposals.times do
+      proposal = create_proposal(requester: user)
+      create_work_order(proposal: proposal)
+      add_comments(proposal: proposal)
+    end
+  end
 
-      # TODO all of these things should have the same created_at/updated_at... use Timecop
-      proposal = FactoryGirl.create(:proposal,
-        :with_serial_approvers,
-        :with_observers,
-        created_at: requested_at,
-        updated_at: requested_at
-      )
+  private
 
-      work_order = FactoryGirl.create(:ncr_work_order,
-        emergency: random_bool(0.1),
-        proposal: proposal,
-        vendor: Faker::Company.name
-      )
+  def create_proposal(requester: create(:user), requested_at: Time.zone.now)
+    create(
+      :proposal,
+      :with_serial_approvers,
+      :with_observers,
+      requester: requester,
+      created_at: requested_at,
+      updated_at: requested_at,
+    )
+  end
 
-      unless work_order.emergency
-        # TODO randomly approve approvals and proposals at different times
```

Comment:
I believe I created the Populator when we were starting work on a reporting dashboard, and having a bunch of Proposals that were approved at the same time wouldn't make for very interesting reports/graphs. The idea with this TODO is that we would add some variance in how far along each Proposal was in the approval process, and how much time passed between each approval. Does that make sense?